pkg/network/tracer/tracer_test.go

```diff
@@ -198,12 +198,13 @@ func (s *TracerSuite) TestTCPSendAndReceive() {
 	require.NoError(t, err)
 
 	var conn *network.ConnectionStats
-	require.Eventually(t, func() bool {
+	require.EventuallyWithT(t, func(collect *assert.CollectT) {
```

Comment:
Does this one need to be changed? I don't see any `require.*` calls instead the `Eventually` loop.

Reply:
yes, as `connections := getConnections(t, tr)` has `require.NoError` in it

Reply:
Worth noting when I tried something like this earlier, I discovered that `getConnections(collect, tr)` will panic because of the require call (not assert) and the resulting panic is mostly unreadable compared to using  `getConnections(t, tr)`. Maybe because `collect` doesn't expect to ever use `require`, not sure. Maybe we need a separate `getConnectionsAssert` or something

Reply:
Just changed `require.NoError` to `require.Error` in `getConnections` just to bring an output of a failure in such a case - 
```bash
=== RUN   TestTracerSuite/CO-RE/TestTCPSendAndReceive
    tracer_test.go:201: 
        	Error Trace:	/home/guy/dd/datadog-agent/pkg/network/tracer/tracer_test.go:973
        	            				/home/guy/dd/datadog-agent/pkg/network/tracer/tracer_test.go:203
        	            				/home/guy/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.23.3.linux-amd64/src/runtime/asm_amd64.s:1700
        	Error:      	An error is expected but got nil.
    tracer_test.go:201: 
        	Error Trace:	/home/guy/dd/datadog-agent/pkg/network/tracer/tracer_test.go:201
        	Error:      	Condition never satisfied
        	Test:       	TestTracerSuite/CO-RE/TestTCPSendAndReceive
        	Messages:   	failed to find connection
        --- FAIL: TestTracerSuite/CO-RE/TestTCPSendAndReceive (5.99s)
```

There's no panic, and the output seems reasonable